src/screens/Customers/AddCustomer.tsx

```diff
@@ -56,52 +58,62 @@ interface AddCustomerProps extends RouteComponentProps {
+    selectedMeterType:
+      yup.string().oneOf([MeterType.NO_METER, MeterType.SMART_METER, MeterType.ANALOG_METER], 'Meter Type must be one of Smart, Analog, or No Meter'),
+    selectedTariffPlanId: yup.string().required('Must select a tariff plan'),
+    meterNumber: yup.mixed().when('selectedMeterType', {
+      is: (MeterType.SMART_METER || MeterType.ANALOG_METER),
+      then: yup.mixed().required('Please enter a meter number')
+    }),
+    customerNumber: yup.number()
+      .min(0, 'Please enter a positive number')
+      .notOneOf(allCustomerNumbers, 'That customer number is used by another customer')
+      .required('Customer number is required')
+  });
 
-  const [customerName, setCustomerName] = useState("");
-  const [customerNumber, setCustomerNumber] = useState("");
-  const [selectedMeterType, setSelectedMeterType] = useState(MeterType.INACTIVE);
-  const [meterNumber, setMeterNumber] = useState(""); // TODO: @julianrkung Look into constraints on meter number input.
-  const [selectedTariffPlanId, setSelectedTariffPlanId] = useState("");
-  const [loading, setLoading] = useState(false);
+  const formik = useFormik({
+    initialValues: {
+      customerName: '',
+      customerNumber: '',
+      selectedMeterType: MeterType.NO_METER,
+      meterNumber: '',
+      selectedTariffPlanId: '',
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      handleSubmit(values);
+    }
+  })
 
+  const [loading, setLoading] = useState(false);
   const currentSite = useSelector(selectCurrentSiteInformation);
   const tariffPlans = useSelector(selectAllTariffPlansArray);
-  tariffPlans.sort(function(a, b) {
+  tariffPlans.sort(function (a, b) {
     return b.numberOfCustomers - a.numberOfCustomers;
   });
 
   if (!currentSite) {
     return <Redirect to={'/customers'} />
   }
 
-  const handleSelectTariffPlan = (event: React.ChangeEvent<{ value: unknown }>) => {
-    setSelectedTariffPlanId(event.target.value as string);
-  }
-
-  const handleSelectMeterType = (event: React.ChangeEvent<{ value: unknown }>) => {
-    setSelectedMeterType(event.target.value as MeterType);
-  }
-
   // TODO: Add form input validation and error messaging
-  const handleSubmit = (event: React.MouseEvent) => {
-
-    // Set button as loading
-    // TODO: error handling
+  const handleSubmit = (values: any) => {
+    const { customerName, customerNumber, selectedMeterType, meterNumber, selectedTariffPlanId } = values;
     setLoading(true);
 
-    // Prevent page refresh on submit
-    event.preventDefault();
-
     // Make a deep copy of an empty customer record
     const customer = JSON.parse(JSON.stringify(EMPTY_CUSTOMER));
     customer.name = customerName;
     customer.customerNumber = parseInt(customerNumber);
     customer.meterType = selectedMeterType;
-    customer.meterNumber = parseInt(meterNumber);
+    customer.meterNumber = isNaN(parseInt(meterNumber)) ? parseInt(meterNumber) : null;
```

Comment:
Confused why you're setting it to the `parseInt` value even when it `isNaN`...mistake?

Reply:
Mistake, good catch, thanks :)